internal/codegen/imports.go

```diff
@@ -99,8 +99,8 @@ func importPathToAssumedName(importPath string) string {
 	}
 	base = strings.TrimPrefix(base, "go-")
 	if i := strings.IndexFunc(base, func(r rune) bool {
-		return !('a' <= r && r <= 'z' || 'A' <= r && r <= 'Z' || '0' <= r && r <= '9' || r == '_' ||
-			r >= utf8.RuneSelf && (unicode.IsLetter(r) || unicode.IsDigit(r)))
+		return (r < 'a' || r > 'z') && (r < 'A' || r > 'Z') && (r < '0' || r > '9') && r != '_' &&
+			(r < utf8.RuneSelf || (!unicode.IsLetter(r) && !unicode.IsDigit(r)))
```

Comment:
Looking at the error in `imports.go`, the issue is in the `ImportFunc` function where De Morgan's law can be applied to simplify a complex negation.

The original code was checking if a character is NOT a valid identifier character by negating a complex OR expression. By applying De Morgan's law:
- `!(a || b)` becomes `!a && !b`
- `!(a && b)` becomes `!a || !b`

The new version directly checks for invalid characters without the negation operator, making the code more readable and maintainable.

This transformation ensures the function's logic remains the same while satisfying the linter rule about applying De Morgan's law to simplify boolean logic.